pkg/providers/cloudstack/reconciler/reconciler.go

```diff
@@ -69,9 +73,28 @@ func (r *Reconciler) ReconcileControlPlane(ctx context.Context, log logr.Logger,
 // ReconcileWorkerNodes validates the cluster definition and reconciles the worker nodes
 // to the desired state.
 func (r *Reconciler) ReconcileWorkerNodes(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
```

Comment:
ReconcileWorkerNodes is Intended to be used on self managed clusters. It's different with `ReconcileWorkers` below. Let's only target `ReconcileWorkers` for this pr